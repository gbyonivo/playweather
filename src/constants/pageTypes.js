export default {
  forecast: 'forecast',
  current: 'current'
};