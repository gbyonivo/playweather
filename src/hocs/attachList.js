import React from 'react';
import LocationListMenu from '../components/locationListMenu';

import styles from './attachList.scss';

export default (Component) => {
  const ListAndReportDetials = props => <div className={styles.listAndReportDetails}>
    <LocationListMenu {...props}/>
    <Component {...props}/>
  </div>;
  return ListAndReportDetials;
};