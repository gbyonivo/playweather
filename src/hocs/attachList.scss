.listAndReportDetails{
  display: grid;
  grid-template-columns: 4fr 1fr;
  grid-gap: 1em;
  grid-gap:10px;
  height: calc(100vh - 100px);
  margin: 10px;
}

@media screen and (max-width: 768px) {
  .listAndReportDetails{
    grid-template-columns: 1fr;
  }
}

@media screen and (max-width: 615px) {
  .listAndReportDetails{
    grid-template-columns: 1fr;
  }
}