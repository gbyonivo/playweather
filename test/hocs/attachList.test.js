import React from 'react';
import renderer from 'react-test-renderer';
import attachList from '../../src/hocs/attachList';
import pageTypes from '../../src/constants/pageTypes';
import Button from '../../src/components/basicElements/button';

const buttonProps = {
  value: 'clickme',
  isLoading: false,
  onClick: () => {}
};

describe('attachList Hoc', () => {
  it('should return component', () => {
    const actual = renderer.create(attachList(<Button {...buttonProps}/>, pageTypes.current));
    expect(actual).toMatchSnapshot();
  });
});