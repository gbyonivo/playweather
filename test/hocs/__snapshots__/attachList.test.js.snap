// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attachList Hoc should return component 1`] = `null`;
